weblab_cg/_generator_chaste_cg.py

```diff
@@ -179,6 +192,30 @@ def _get_var_display_name(self, var):
             display_name = var.cmeta_id
         return display_name.replace('$', '__').replace('var_chaste_interface_', '', 1).replace('var_', '', 1)
 
+    def _partial_eval(self, equations, required_lhs):
```

Comment:
One thing I learnt through doing pycml was to move these model analysis/transformation steps into their own class/module, rather than embedded in other parts of the code. So I'd be inclined to create a new module to hold this - it's just a standalone function after all! Not sure what you'd call it; guess it depends on whether lookup tables goes in the same file, and where things like rush-larsen end up going. You might want a sub-package for all analysis/transformations.

Reply:
Actually good point I realized I forgot to do it, but it was exactly my intention when I created the OO structure, now addressed.